quickwit/quickwit-metastore/src/metastore/postgres/metastore.rs

```diff
@@ -1310,16 +1389,21 @@ impl MetastoreService for PostgresqlMetastore {
             .bind(request.force)
             .fetch_all(&self.connection_pool)
             .await?;
-        if !request.force
-            && pg_shards.into_iter().any(|pg_shard| {
-                let position: Position = pg_shard.publish_position_inclusive.into();
-                position.is_eof()
-            })
+
+        if request.force {
+            return Ok(EmptyResponse {});
+        }
+        if let Some(pg_shard) = pg_shards
+            .into_iter()
+            .find(|pg_shard| Position::is_eof_str(&pg_shard.publish_position_inclusive))
         {
-            let message = "failed to delete shard ``: shard is not fully indexed".to_string();
+            let message = format!(
+                "failed to delete shard `{}`: shard is not fully indexed",
+                pg_shard.shard_id
```

Comment:
I don't quite understand what is the `pg_shards`  varialbe about and even more what this `if let` statement is about.

Isn't the DELETE statement without `RETURNING` returning the number of modified rows?

Assuming this was returning actual deleted rows, why do we check for them to be EOF and log an error message saying otherwise?

Reply:
Yes, this is trash.